lib/spack/spack/test/architecture.py

```diff
@@ -210,8 +209,8 @@ def test_optimization_flags_with_custom_versions(
     target = spack.architecture.Target(target_str)
     if real_version:
         monkeypatch.setattr(
-            spack.compiler.Compiler, 'cc_version', lambda x, y: real_version
-        )
+            spack.compiler.Compiler, 'compute_real_version',
+            lambda x: real_version)
```

Comment:
is the purpose of "real version" documented somewhere?  If not, it should be.  I think that can be done generically by just saying it's to make up for bad versions in `compilers.yaml`.

Reply:
I think this is adequately documented where the method is called in `architecture.py`